### DISCUSSION TITLE
Changes and challenges with the new version of Selling Partners API

### DISCUSSION THREAD

**Discussion by gabefishbein** [amzn/selling-partner-api-models#606] [Ideas]
I'm on the product team for a company who would be considered a '3rd party developer' in Amazon's terminology and who builds software that is used by hundreds of Amazon vendors and 3rd party sellers. Our company has been in business with Amazon for a decade. Since 2021, we've been excited by the release of new Selling Partner API report types, specifically those that work for vendors, as this type of selling partner has been historically underserved when it came to Amazon APIs and is a critical customer segment for my company. Since the March 2022 SP-API release announcement we've been exploring the new versions of the 'vendor retail analytics reports' and have some constructive feedback to share. By sharing this feedback in a public forum, we're hoping it will resonate with the broader developer community and help Amazon appropriate prioritize addressing the feedback.

Data latency matters. A lot. The speed at which our clients (predominantly Amazon vendors) can get up-to-date data on how their Amazon business is performing is critical to inform a variety of operational and marketing decisions, including whether or not to allocate more inventory to Amazon, start, stop, or adjust a sponsored advertising campaign, and run a deal or coupon. Until now, we've come to expect a latency of about 2 days, meaning we could measure Monday's data by Thursday. This wasn't ideal, but was workable. With the new versions of the retail analytics reports this latency has increased to ~5 days (and is advertised as being as long as 7 days!). We'd like Amazon to invest in dropping the latency to as short as 6 hours, so that our clients can wake up on Tuesday and see what happened on Monday. We see this especially important during events such as Prime Day, Black Friday, Cyber Monday, and other times we're running deals, coupons, etc.

Another point: where did daily views of inventory levels go? The previous versions of inventory health reports enabled vendors to get a view into Amazon's inventory position on their own products at the 'day' resolution. The new reports only give vendor the 'week' resolution. This degradation in resolution negatively impacts vendor decisions across the board, both from operations and marketing perspectives. For example, how would a vendor know to ramp up supply for Amazon on a fast-moving item mid-week if their only view into Amazon's inventory position is as of last week when inventory levels seemed fine? The same scenario could apply in the opposite case, where Amazon's inventory levels improve mid-week but the vendor doesn't know this until 3 days later and has lost the opportunity to resume advertising the item during the period. With Amazon typically keeping single-digit weeks of supply on hand, it's imperative we're able to see at the daily level.

Finally, we _really_ need OOS and LBB metrics; please bring those back! Previously, we used these metrics to inform vendors of any supply issues that need resolution, as well as any opportunity to improve pricing. Without knowing if we're losing buy box or if we're out of stock, we lose the ability to ensure a tight supply chain. Coupled with losing high grain inventory data (above), we're flying more blind than ever when supply chain challenges have never been more prevalent for many of the vendors that we manage.

I imagine there are reasons for these changes, but we're hopeful that Amazon will use this information to inform their roadmap. To quote [Jeff Bezos' 2016 Annual Shareholder Letter](https://www.aboutamazon.com/news/company-news/2016-letter-to-shareholders), we want to empower our clients to make high-quality, _high-velocity_ decisions, and we'd love the data speed and fidelity to do so.

**Comment by gabefishbein** [comment]
Some additional feedback we've gotten from a _very_ high volume vendor on Amazon around their expectations for data availability. We're stuck in a place where we have to tell them their ask is not possible, despite the request being fairly reasonable. See their ask below:

> So to be clear we’d need a report delivered to us every Sunday night, showing the shipped sales, by day, for all units sold in the previous Sunday – Saturday inclusive. We’d then need a separate report on the Sunday night showing inventory levels on all skus on the Sunday please. Could you confirm if this is possible please? 

> This cadence challenge could mean that we’re unable to proceed and indeed could pose a larger problem with our AMZ engagement. Currently we get the data we need on a Monday for the preceding week’s sales and inventory. We then digest this and use it for insight and adjustments to our planning. Our challenge is to move from weekly to daily info at a speedier cadence thus enabling our teams to focus on driving our business rather than reporting it.

**Reply by psneffem** [reply]
**Data latency matters. A lot.**
This cannot be understated.

I work in a large FMCG organization, with substantial sales and investments (+JBPs) with Amazon across categories and countries. We also have invested strongly in data and analytics, with Amazon data feeding multiple streams of insight and action.

We use Amz APIs and the UIs to improve decision making in the service of our customers (Amazon customers). The investment in amz and in our technology infrastructure is directly linked to the speed and quality (+ quantity) of data amz provide. The API was keenly greeted, as it allowed for a sustainable data flow, away from the UI.

I’m pressed internally on the 2 day delay currently – as mentioned in the above post – I too would like it to be quicker. The delay on the API has seen us rely more strongly (again) on the UI (the problem we were solving with the API) – if the latency in the API is mirrored in the UI, we would have a significant business challenge. This would impact investment decisions across Amazon and in a period when budgets are being scrutinized everywhere – re-open discussions on the basis of data availability. 

All this to say – the changes outlined above – which were supporting decisions in Amazon are hobbled by latency in the API, they will be crippling if latency migrates to the UI.

Please can you provide clarity as to whether this is a teething problem? One which will be resolved, hopefully swiftly? Or if this marks a broader change.

**Comment by sungolivia** [comment]
Hey @gabefishbein @psneffem even though this thread is bit older, our team has posted [vendor reports FAQ ](https://developer-docs.amazon.com/sp-api/page/sp-api-for-vendors-reports-faq)which you might have already seen before. 

I heard a lot about the same issues that is presented in this thread and our team is aware. 
Hopefully I can share some better news in the future and will be updating this thread as well, but thanks again for the feedback, keep it up!

**Reply by gabefishbein** [reply]
@sungolivia , are you able to point to which FAQs address the data SLA question that @psneffem and I have? Of all, I think this is most critical and performance seems to be worsening over past few weeks (now slipping to 4-5 day lag). 

Thank you much.
